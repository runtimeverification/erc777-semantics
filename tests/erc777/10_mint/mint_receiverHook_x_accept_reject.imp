test.preamble();

// 2 receiver addresses, one accepting another rejecting.
//    Mint by one operator from 1st, then 2nd, then 1st.

test.registerTokensReceivedHook(5, 1); //accepting
test.registerTokensReceivedHook(6, 3); //rejecting
test.makeCaller(8);
test.operatorMint(5, 1000, "", "");
test.operatorMint(6, 10000, "", "");
test.operatorMint(5, 100000, "", "");
