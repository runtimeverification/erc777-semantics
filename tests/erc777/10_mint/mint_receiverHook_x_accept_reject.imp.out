Registering receiver 5 with hook 1
Registering receiver 6 with hook 3
Changing the caller to 8
8 is minting 1000 for 5 (data:""),  (operatorData:"")
tokensReceived: Hook address = 1, Operator = 8, From = 0, To = 5, Value = 1000, Data = , OperatorData = 
tokensReceived: sender balance: 0
tokensReceived: receiver balance: 1050
Balance of 5 is 1050
Total supply is 1550
8 is minting 10000 for 6 (data:""),  (operatorData:"")
tokensReceived: Hook address = 3, Operator = 8, From = 0, To = 6, Value = 10000, Data = , OperatorData = 
tokensReceived: sender balance: 0
tokensReceived: receiver balance: 10060
Reverting...
REVERT
Balance of 6 is 60
Total supply is 1550
8 is minting 100000 for 5 (data:""),  (operatorData:"")
tokensReceived: Hook address = 1, Operator = 8, From = 0, To = 5, Value = 100000, Data = , OperatorData = 
tokensReceived: sender balance: 0
tokensReceived: receiver balance: 101050
Balance of 5 is 101050
Total supply is 101550
